test/providers/summary/scancode-new.js

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) Microsoft Corporation and others. Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+
+const assert = require('assert')
+const summarizer = require('../../../providers/summary/scancode-new')()
+summarizer.logger = { info: () => {} }
+const fs = require('fs')
+const path = require('path')
+const { get, uniq, flatten } = require('lodash')
+const { expect } = require('chai')
+
+const scancodeVersions = ['32.0.8']
+
+describe('ScancodeSummarizerNew basic compatability', () => {
+  it('summarizes basic npm', () => {
+    const coordinates = { type: 'npm', provider: 'npmjs' }
+    for (let version of scancodeVersions) {
+      const harvestData = getHarvestData(version, 'npm-basic')
+      if (!harvestData) continue
+      const result = summarizer.summarize(coordinates, harvestData)
+      assert.equal(result.licensed.declared, 'ISC', `Declared license mismatch for version ${version}`)
+      assert.equal(result.described.releaseDate, '2017-05-19', `releaseDate mismatch for version ${version}`)
+      assert.deepEqual(uniq(flatten(result.files.map((x) => x.attributions))).filter((x) => x).length, 1)
+      assert.deepEqual(result.files.find((x) => x.path === 'package/LICENSE').natures, ['license'])
+      assert.equal(flatten(result.files.map((x) => x.natures)).filter((x) => x).length, 1)
+    }
+  })
+
+  it('summarizes large npm', () => {
+    const coordinates = { type: 'npm', provider: 'npmjs' }
+    const overrides = {
+      // TODO: Verify correctness of the 32.0.8 override
+      '32.0.8': 'BSD-3-Clause AND GPL-2.0-only AND GPL-1.0-or-later AND BSD-3-Clause AND GPL-2.0-only'
```

Comment:
@yashkohli88 Could you please provide input from an compliance analyst point of view?  The package is npm/npmjs/-/node-forge/0.7.5.

Reply:
From compliance analyst point of view, the accurate declared license for this component would be dual license between "BSD 3-Clause License OR GNU General Public License Version 2". 

This can be verified using https://github.com/digitalbazaar/forge/blob/c3d63fbbee1e9f68108bd499ceaeb01b5800e95c/LICENSE#L1

Reply:
The matched text corresponds to "**[If the GPL suits your project better you are also free to use Forge under
that license.](https://github.com/digitalbazaar/forge/blob/2bb97afb5058285ef09bcf1d04d6bd6b87cffd58/LICENSE#L8-L9)**" And I can verify from the license file that all other occurrences of GPL keyword are accompanied with its version i.e. version 2. Hence the matcher marks GPL 1 plus because of this line. But in my opinion scancode should not flag this as GPL1 plus (as this would mean it could be either version 2 or version 3) just on the basis of this line and especially when it is detecting the exact version of license which does not include any 'later' clause.

Reply:
Reported via https://github.com/nexB/scancode-toolkit/issues/3722